code/modules/reagents/chems/chems_ethanol.dm

```diff
@@ -36,6 +40,12 @@
 	glass_desc = "A well-known alcohol with a variety of applications."
 	value = 1.2
 
+/decl/material/liquid/ethanol/Initialize()
+	. = ..()
+	// Impure ethanol doesn't boil, it has to separate first.
+	if(type != bypass_heating_products_for_root_type)
```

Comment:
`bypass_heating_products_for_root_type` is only ever used for ethanol, and presumably it exists to avoid exactly this initialize override, so it could probably be changed to just a bespoke `type != /decl/material/liquid/ethanol` check that just unsets all the relevant vars here. idk, just a nitpick, i don't really like that this (and the chilling products counterpart) is so niche and only ever used for ethanol...

feel free to ignore this one, though, just idle musing